packages/SwingSet/src/controller.js

```diff
@@ -125,15 +125,48 @@ export function loadBasedir(basedir) {
   return config;
 }
 
+/**
+ * Resolve a pathname found in a config descriptor.  First try to resolve it as
+ * a module path, and then if that doesn't work try to resolve it as an
+ * ordinary path relative to the directory in which the config file was found.
+ *
+ * @param dirname  Path to directory containing the config file
+ * @param specPath  Path found in a `sourceSpec` or `bundleSpec` property
+ *
+ * @return the absolute path corresponding to `specPath` if it can be
+ *    determined.
+ */
+function resolveSpecFromConfig(dirname, specPath) {
+  try {
+    return require.resolve(specPath);
```

Comment:
If this is given `./vats/vat-timerWrapper.js`, will it return the absolute path of `SwingSet/src/vats/vat-timerWrapper.js` instead of something near the config file? That seems surprising. I'm inclined to only call `require.resolve` here when `specPath` doesn't start with `./` or `../`.

Reply:
Please investigate using:

```js
function resolveSpecFromConfig(dirname, specPath) {
 // This is only resolved by Node's algorithm relative to dirname.
 return require.resolve(specPath, { paths: [dirname] });
}
```

This trick works for all absolute and relative paths, even outside of Node.js packages.  No need to be clever when Node.js already does the work for you.

Reply:
Please see my top-level comment on this PR.

Reply:
Giving `require.resolve` a path eliminates the need to inspect the path to see if it's relative to `'./'` or `'../'` takes care of the case that Brian flagged without the need to inspect the path name string directly, but by itself it's not sufficient. It will produce a MODULE_NOT_FOUND error for a bare filename (e.g., `foo.js`) or a path beginning with a directory name (e.g., `subdir/foo.js`), so you still need to do a `path.resolve` to handle those cases properly.